docs: updates of ctags(1) and docs/ updates

hirooih opened the pull request:
- ctags(1) updates
  - markups fixes
    - 5c473165 docs: ctags(1): more consistent markups
    - 0da01779 docs: ctags(1): not use Sphinx extensions
    - 04e8a42c docs: ctags(1): additional markup fixes
  - contents reordering
    - e0fd0467 docs: ctags(1): reorder tag entries related descriptions
  - contents fixes (**please review carefully**)
    - 92a648ca docs: ctags(1): misc fixes
    - 2f6f62bd docs: ctags(1): update TAGS FILE FORMAT and TAG ENTRIES
    - dba2d592 docs: ctags(1): update OPTIONS
- docs fixes
  - db6aec7a docs: update and deal with TODO list

masatake commented on docs/man/ctags.1.rst:
```diff
@@ -214,25 +219,33 @@ Input/Output File Options
+	option includes this option.
+
+.. TODO: Does not --list-features option support this?
+
```
Thank you very much. This is an important suggestion. How do you think about introducing `.. TODO(code):` markup for the suggestions for improving the program code, not for the documents? So I can pick up this.

masatake commented:
Could you split the pull request into two?
```

    5c47316 docs: ctags(1): more consistent markups
    0da0177 docs: ctags(1): not use Sphinx extensions
    04e8a42 docs: ctags(1): additional markup fixes
```

I would like you to merge this after updating man/README.
Could you consider and fix the markup rule for the man pages and write it to man/README.
We, the ctags developers recognize you as the maintainer of man/README.
The developers must get an ack from you when changing man/README.

The README is needed to write consistent man pages. I add a man page in #2844.
I wonder whether the new man page satisfies the rules that will be written in the README.

Thanks in advance.

hirooih commented:
@masatake san,

Thank you for your feedbacks.

> Could you split the pull request into two?

I also wanted to do that, but it is very very hard.

>    5c47316 docs: ctags(1): more consistent markups

This is the first commit in this pull-request.

> docs: ctags(1): reorder tag entries related descriptions …

This is the third commit.  I reordered the contents aggressively.  This made it very hard to split the pull request.

>    0da0177 docs: ctags(1): not use Sphinx extensions
>    04e8a42 docs: ctags(1): additional markup fixes

I admit that these last two commits are my omissions (TENUKI, in Japanese).
I've squashed 0da0177 into "2f6f62b docs: ctags(1): update TAGS FILE FORMAT and TAG ENTRIES", and 04e8a42  into "5c47316 docs: ctags(1): more consistent markups".

I hope you accept this.

***

I am already updating man/README.  I will send another pull-request soon.
After that I am happy to propose update of the help message.

***

> How do you think about introducing .. TODO(code): markup ...

It is good idea.  I've fixed the comment.
I will also add this rule in man/README.

***

> [jet@living]~/var/ctags-new/man% echo --./*
> echo --./*
> zsh: no matches found: --./*

First I thought this was a warning message caused by your personal setting for zsh.  But I found that it is an error, and was the 2nd significant incompatibility "features" of zsh.

[Z-Shell Frequently-Asked Questions, 2.1: Differences from sh and ksh](http://zsh.sourceforge.net/FAQ/zshfaq02.html#l10)

> The next most classic difference is that unmatched glob patterns cause the command to abort; set NO_NOMATCH for those.

zsh users should know about this feature.  Search "zsh: no matches found", then you can find many questions about this error.  I think ctags(1) man page should not take care of it.  When I made the ctags-faq(7), this is one of the candidates to add to it.

> Anyway, this change is nothing about 'not use Sphinx extensions '.

It has been in "docs: ctags(1): update OPTIONS", since I committed last night. I think it is a proper commit for this.

masatake commented on docs/man/ctags.1.rst:
```diff
@@ -773,15 +806,17 @@ Tags File Contents Options
 	are useful for testing.
 
+	.. TODO: "to all languages;" instead of "to all fields;" ???
+
```
```
If specifying nothing as the parameter ``flags``, all fields are disabled.
e.g. ``--fields-C=`` means disabling all language-specific fields of C.
``--fields-all=`` means disabling all language-specific fields of all languages.
```

masatake commented:
Does TODO mean your TODO, our TODO, or a question to @mastake?

I want to answer your questions. So I would like you to use special markers for this purpose.
I guess, in some cases, you have questions, but you don't need answers for them now because you will not reflect the answers to this pull request.

So, how about using the following markers: `Q(now)`, and `Q`.

I will answer for `Q(now)` while reviewing this pull request.

`Q(now)` is just an idea. You can use a better marker for the questions that @mastake should answer now.

masatake commented:
My review board
- IN PROGRESS
  - docs: ctags(1): update TAGS FILE FORMAT and TAG ENTRIES 
- LGTM
  - docs: update and deal with TODO list 
  - docs: ctags(1): update OPTIONS

hirooih commented:
@masatake san,

Thank you for sharing the review status.  Please take your time.

> Does TODO mean a documentation TODO or a question to @mastake?

As you may have noticed, it is case by case.  I should clarify my intentions more. I will try that in my future pull-requests.

>	.. TODO: "to all languages;" instead of "to all fields;" ???

In this case this was a question to myself.  I found that the original description was correct and I removed the TODO comment in "docs: ctags(1): update OPTIONS".

Now I find that I was able to squash "docs: ctags(1): update OPTIONS" to "docs: ctags(1): misc fixes".  These commits are not affected by reordering in "docs: ctags(1): reorder tag entries related descriptions".

masatake commented on docs/man/ctags.1.rst:
```diff
@@ -1565,28 +1548,47 @@ Examples:
-		vi(1) implementations.
+		``vi(1)`` implementations.
+
+A few special tags, called *pseudo tags*, are written into the tag file for internal purposes.
```
```
are written into the tag file for explaining the tag file itself.
They are started from `!_`.
````

masatake commented on docs/man/ctags.1.rst:
```diff
@@ -1721,26 +1717,41 @@ The meaning of major fields is as follows (one-letter flag/long-name flag):
+
+.. code-block:: console
+
+	$ ctags -o - kinds.c
```
Nice!

masatake replied:
It will be nice having similar examples for `scope:`.

masatake replied:
These are just comments. Reflecting on the comment to this pull request is not a must.

hirooih replied:
> It will be nice having similar examples for scope:.

That is a good idea.  I will add it in another pull-request.

masatake commented on docs/man/ctags.1.rst:
```diff
@@ -1868,40 +1917,63 @@ none/``anonymous``
+		X       inc.c   /^#define X$/;" macro   file:   roles:def       extras:fileScope
+		X       inc.c   /^#undef X$/;"  macro   file:   roles:undef     extras:fileScope,reference
+
 Language-specific fields and extras
```
TEMPORARY: MY BOOKMARK

masatake commented on docs/man/ctags.1.rst:
```diff
@@ -686,12 +684,6 @@ Tags File Contents Options
 ``--<LANG>-kinds=[+|-]kinds|*``
 	This option is obsolete. Use ``--kinds-<LANG>=...`` instead.
 
-``--param-<LANG>:name=argument``
```
Could you keep ``--param-<LANG>`` as a comment?
Another contributor may think why there is no description for the option.
Keeping this as a comment may help the person understand the situation.

masatake commented on docs/man/ctags.1.rst:
```diff
@@ -1342,7 +1415,7 @@ Listing Options
 		Whether the kind is enabled or not. It takes ``yes`` or ``no``.
-		(Currently all roles are enabled. No option for disabling
+		(Currently all roles are enabled. An option for disabling
 		a specified role is not implemented yet.)
```
I remember  I implemented an option for enabling/disabling roles:
```
--roles-<LANG>.kind=[+|-]role
```

hirooih replied:
I've removed the note of the restriction.

masatake commented on docs/man/ctags.1.rst:
```diff
@@ -1512,7 +1589,7 @@ all files in an entire source directory tree
 names are mapped to languages will be scanned.
 
-The same extensions are mapped to multiple parsers. For example, ``.h``
+The an extension may be mapped to multiple parsers. For example, ``.h``
```
I think `The' is not needed here.

hirooih replied:
Whoops!  FIxed.

masatake commented:
@hirooih, thank you very much. This is great work.
I wrote some minor comments.
Other than them, the changes look good to me.

hirooih commented:
@masatake san,
Thank you for your review.

> @hirooih, thank you very much. This is great work.

You are so welcome.
I also thank you for your great work on this project.

I will work for the examples for scope in another pull-request.  It needs a new subsection.
I fixed other suggestions.
